core/interfaces/rebase.py

```diff
@@ -284,6 +296,95 @@ def base_ref(self):
 
         return base_ref
 
+    def nearest_branch(self, branch, default="master"):
```

Comment:
It's probably worth breaking this out into a helper method, since there's some complexity here and it isn't strictly view-related.